File: python/privacy_kit/ropa.py
"""Genera el Registro de Actividades de Tratamiento (ROPA) desde el config."""
from __future__ import annotations
from .config import PrivacyConfig


def build_ropa(config: PrivacyConfig) -> str:
    r = config.raw
    lines = ["# Registro de Actividades de Tratamiento (ROPA)", ""]
    resp = r.get("responsable", {})
    lines += [f"**Responsable:** {resp.get('nombre')} ({resp.get('rut')})",
              f"**Contacto privacidad:** {resp.get('contacto_privacidad')}", ""]
    lines.append("## Finalidades y base de licitud")
    for p, meta in r.get("purposes", {}).items():
        lines.append(f"- **{p}** — base: {meta.get('legal_basis')}")
    lines.append("\n## Categorías de datos y retención")
    for cat, fields in r.get("data_categories", {}).items():
        dias = r.get("retention_days", {}).get(cat, 0)
        lines.append(f"- **{cat}** ({', '.join(fields) or '—'}) — retención: {dias} días")
    lines.append("\n## Transferencias a terceros")
    for name, meta in r.get("third_parties", {}).items():
        lines.append(f"- **{name}** — país: {meta.get('pais')}, rol: {meta.get('rol')}, base: {meta.get('base')}")
    return "\n".join(lines) + "\n"

File: node/src/ropa.js
'use strict';
// Genera el Registro de Actividades de Tratamiento (ROPA) desde el config.
function buildRopa(config) {
  const r = config.raw;
  const lines = ['# Registro de Actividades de Tratamiento (ROPA)', ''];
  const resp = r.responsable || {};
  lines.push(`**Responsable:** ${resp.nombre} (${resp.rut})`,
             `**Contacto privacidad:** ${resp.contactoPrivacidad}`, '');
  lines.push('## Finalidades y base de licitud');
  for (const [p, m] of Object.entries(r.purposes || {})) lines.push(`- **${p}** — base: ${m.legalBasis}`);
  lines.push('', '## Categorías de datos y retención');
  for (const [cat, f] of Object.entries(r.dataCategories || {})) {
    const d = (r.retentionDays || {})[cat] || 0;
    lines.push(`- **${cat}** (${(f || []).join(', ') || '—'}) — retención: ${d} días`);
  }
  lines.push('', '## Transferencias a terceros');
  for (const [n, m] of Object.entries(r.thirdParties || {}))
    lines.push(`- **${n}** — país: ${m.pais}, rol: ${m.rol}, base: ${m.base}`);
  return lines.join('\n') + '\n';
}
module.exports = { buildRopa };
